### PR TITLE
Refugee Expansion and Proficiency Project

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
SUMMARY: Content "Expands and improves refugee quests, aim to get them to teach you proficiencies"

#### Purpose of change
I've left the back room refugees a bit abandoned, which is thematically appropriate given the free merchants' attitudes towards them.

Now that we have proficiencies, we need some proficiency teachers. This is a good way to encourage me to get some work done on these poor dudes.

#### Describe the solution
- [x] Add quests to help Boris deal with his grief
- [x] Get boris to teach you carpentry
- [x] Pay fatima to get her to teach you welding
- [x] Barter with Dana about bread prices
- [x] Jenny to teach you soldering
- [x] Add a counter to the overall "refugee hapiness" that ticks up as you do things to help them
- [x] add dialogue dependent on refugee happiness variable
- [x] Uyen can teach you first aid (and knitting). Will eventually get paramedic-related proficiencies when they exist.

Postponed for another PR because this is getting big:
- Expand Jenny's and Boris' quests to include update_mapgen,
- Add mission dialogue to all refugees related to missions where you have to ask around (mole mission)
- [change dialogue-only missions among beggars to be proper missions
- Create an overarching checklist mission to recruit all the beggars 
- Add more dialogue guidance on conditions needed to get beggars recruited

In the process, begin the stuff to clear out the back bay and have the refugees move in there (although I am unsure if they can actually 'move' yet).

#### Describe alternatives you've considered
The dialogue and choices of proficiencies could be different of course.

#### Testing
Loaded, it ran, haven't yet tested the dialogue linkages.

#### Additional context
Dana would teach you baking, and Jenny would teach you engineering and design, but those proficiencies don't exist yet.

### CODE REVIEW

**Comment by mlangsdorf** [comment]
NPCs can't move, but ping me about this after 0.F and I'll add a "teleport_to" and "walk_to" set of dialogue effects.

**Comment by Xaleth** [comment]
Can't the NPCs know multiple proficiencies? Since they can only teach you one, they're like static entities

**Comment by I-am-Erk** [comment]
Who says they can only teach you one?

**Comment by I-am-Erk** [comment]
I'm not quite done, but enough of this is functional now that I can take it out of draft. still needs playtesting and I would like to give Jenny the ability to teach you a bit.

File: data/json/npcs/refugee_center/surface_refugees/NPC_Aleesha_Seward.json [hunk 1]
```diff
@@ -151,14 +165,5 @@
       { "text": "What were you saying before?", "topic": "TALK_NONE" },
       { "text": "I'd better get going.", "topic": "TALK_DONE" }
     ]
-  },
-  {
-    "type": "talk_topic",
-    "id": "TALK_REFUGEE_Aleesha_Situation",
-    "dynamic_line": "I don't have much of a hot take besides what I just said.  It's shit.  Not enough to do to stay busy, not enough to eat, and even if I had internet, all the YouTubers I follow are dead.",
```
Comment (LaVeyanFiend) [inline data/json/npcs/refugee_center/surface_refugees/NPC_Aleesha_Seward.json:158]: image
Is there meant to be a replacement line in this PR?

Comment (I-am-Erk) [reply data/json/npcs/refugee_center/surface_refugees/NPC_Aleesha_Seward.json]: Nope, that was an accidental delete. thanks

**Comment by Maleclypse** [comment]
Are you still looking for playtesters on this?

**Pull request merged**